connector/routingconnector/router.go

```diff
@@ -28,8 +29,9 @@ type consumerProvider[C any] func(...pipeline.ID) (C, error)
 // parameter C is expected to be one of: consumer.Traces, consumer.Metrics, or
 // consumer.Logs.
 type router[C any] struct {
-	logger *zap.Logger
-	parser ottl.Parser[ottlresource.TransformContext]
+	logger         *zap.Logger
+	resourceParser ottl.Parser[ottlresource.TransformContext]
+	logParser      ottl.Parser[ottllog.TransformContext]
```

Comment:
Would it make sense to add an additional type parameter for the signal specific parser (or the `TransformContext`). That way when we add support for traces and metrics we don't end up with two more parsers. If the answer yes, but not now, I'd be fine documenting that decision somewhere.

Reply:
We may need to create and manage both, because each route may use a different context. I suppose we might be able to abstract it so that we're managing them in a map but that seems more complicated than just having direct access.